ansible/configs/kni-osp/software.yml

```diff
@@ -0,0 +1,264 @@
+        mode: 0744
+        force: yes
+
+    - name: Resolving templated scripts 4/8
+      template:
+        src: "./files/08_grabber.sh.j2"
+        dest: "/home/cloud-user/scripts/08_grabber.sh"
+        owner: "cloud-user"
+        mode: 0744
+        force: yes
+
+    - name: Resolving templated scripts 5/8
+      template:
+        src: "./files/09_deploy.sh.j2"
+        dest: "/home/cloud-user/scripts/09_deploy.sh"
+        owner: "cloud-user"
+        mode: 0744
+        force: yes
+
+    - name: Resolving templated scripts 6/8
+      template:
+        src: "./files/install-config.yaml.j2"
+        dest: "/home/cloud-user/scripts/install-config.yaml"
+        owner: "cloud-user"
+        mode: 0744
+        force: yes
+
+    - name: Resolving templated scripts 7/8
+      template:
+        src: "./files/08_build_installer.sh.j2"
+        dest: "/home/cloud-user/scripts/08_build_installer.sh"
+        owner: "cloud-user"
+        mode: 0744
+        force: yes
+
+    - name: Resolving templated scripts 8/8
+      template:
+        src: "./files/10_volume-attach.sh.j2"
+        dest: "/home/cloud-user/scripts/10_volume-attach.sh"
+        owner: "cloud-user"
+        mode: 0744
+        force: yes
+
+    - name: Templating dhcp
+      template:
+        src: "./files/dhcpd.conf.j2"
+        dest: "/home/cloud-user/scripts/dhcp/dhcpd.conf"
+        owner: "cloud-user"
+        mode: 0744
+        force: yes
+
+    - name: Templating DNS 1/3
+      template:
+        src: "./files/0.20.10.in-addr.arpa.j2"
+        dest: "/home/cloud-user/scripts/named/0.20.10.in-addr.arpa"
+        owner: "cloud-user"
+        mode: 0744
+        force: yes
+
+    - name: Templating DNS 2/3
+      template:
+        src: "./files/named.conf.j2"
+        dest: "/home/cloud-user/scripts/named/named.conf"
+        owner: "cloud-user"
+        mode: 0744
+        force: yes
+
+    - name: Templating DNS 3/3
+      template:
+        src: "./files/myzone.com.zone.j2"
+        dest: "/home/cloud-user/scripts/named/{{ my_ocp_domain }}.zone"
+        owner: "cloud-user"
+        mode: 0744
+        force: yes
+
+    - name: Setting pull-secret
+      template:
+        src: "./files/pull-secret.json.j2"
+        dest: "/home/cloud-user/pull-secret.json"
+        owner: "cloud-user"
+        mode: 0744
+
+    - name: Sending wrapper script
+      template:
+        src: "./files/wrapper.sh.j2"
+        dest: "/home/cloud-user/scripts/wrapper.sh"
+        owner: "cloud-user"
+        mode: 0744
+
+    - name: Installing KNI if my_auto enabled, this may take a while ...
+      shell: /home/cloud-user/scripts/wrapper.sh
+      args:
+        chdir: /home/cloud-user/scripts/
+
+    - name: Preparing provisionning 7/8
+      command: "/home/cloud-user/scripts/cleanup-refresh.sh"
+      become_user: cloud-user
+
+    - name: Dump all vars Software
+      action: template src=./files/dumpvars.j2 dest=/tmp/ansible-sw.all
+
+- name: Step 003 Bastion VM config
```

Comment:
if you do not update all packages from common role in pre-software, please do a yum update here as well.